lib/bap/bap_log.ml

```diff
@@ -54,13 +54,26 @@ let print_message ppf msg =
   | Error -> eprintf "%s@\n%!" msg.message
   | _ -> ()
 
+let lock log_folder =
+  let lock = log_folder / "lock" in
+  let lock = Unix.openfile lock Unix.[O_RDWR; O_CREAT] 0o666 in
+  Unix.lockf lock Unix.F_LOCK 0;
+  lock
+
+let unlock lock =
+  Unix.lockf lock Unix.F_ULOCK 0;
+  Unix.close lock
+
 let open_log_channel user_dir =
   try
     let log_folder = log_folder user_dir in
     mkdir log_folder;
     let file = log_folder / "log" in
-    if Sys.file_exists file
-    then rotate max_logs file;
+    let lock = lock log_folder in
```

Comment:
In fact, it looks like that the `mkdir log_folder` expression is also experiencing the same TOCTOU issue, so could you please protect everything from the line 70.